src/main/java/com/openshift/internal/restclient/model/BuildConfig.java

```diff
@@ -77,6 +79,38 @@ public BuildConfig(ModelNode node, IClient client, Map<String, String []> proper
 	}
 
 	@Override
+	public void addBuildTrigger(IBuildTrigger trigger) {
+		ModelNode triggers = get(BUILDCONFIG_TRIGGERS);
+		ModelNode triggerNode = triggers.add();
+		switch(trigger.getType()) {
+		case generic:
+			if(!(trigger instanceof IWebhookTrigger)) {
+				throw new IllegalArgumentException("IBuildTrigger of type generic does not implement IWebhookTrigger");
+			}
+			IWebhookTrigger generic = (IWebhookTrigger)trigger;
+			triggerNode.get(new String[] {"generic", "secret"}).set(generic.getSecret());
+			break;
+		case github:
+			if(!(trigger instanceof IWebhookTrigger)) {
+				throw new IllegalArgumentException("IBuildTrigger of type github does not implement IWebhookTrigger");
+			}
+			IWebhookTrigger github = (IWebhookTrigger)trigger;
+			triggerNode.get(new String[] {"github", "secret"}).set(github.getSecret());
+			break;
+		case imageChange:
```

Comment:
@aslakknutsen these work for now, but the back-end has moved to object refs which is simpler and only requires you to reference an instance of a kind by name.  Don't quote me, but I think its the name of an imagestream